text type hint in LinkButtonElement does not accept PlainTextObject/dict

### Reproducible in:

The following code results in a type checking error (using `mypy` for type checking)

```python
 LinkButtonElement(
    text=PlainTextObject(
        text="text    
    ),
    url="http://slack.com",
)
```

```
main.py:551: error: Argument "text" to "LinkButtonElement" has incompatible type "PlainTextObject"; expected "str"
```

From what I understand, a PlainTextObject is appropriate for this element? https://api.slack.com/reference/block-kit/block-elements#button__examples

Could you please confirm whether 1 & 2 below are both valid
1.
```python
 LinkButtonElement(
    text=PlainTextObject(
        text="text    
    ),
    url="http://slack.com",
)
```
2.
```python
 LinkButtonElement(
    text="text"
    url="http://slack.com",
)
```
#### The Slack SDK version

```
slack-sdk==3.14.1
```

#### Python runtime version

```
Python 3.9.6
```

#### OS info

```
ProductName:	macOS
ProductVersion:	12.2
BuildVersion:	21D49
Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64
```

#### Steps to reproduce:

1. Create a LinkButtonBlock with a PlainTextObject element
2. run mypy on your code

### Expected result:

I expect to see no error from mypy

### Actual result:

```
main.py:551: error: Argument "text" to "LinkButtonElement" has incompatible type "PlainTextObject"; expected "str"
```

### Requirements

For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/python-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

--- a/slack_sdk/models/blocks/block_elements.py
+++ b/slack_sdk/models/blocks/block_elements.py
@@ -325,7 +325,7 @@ class LinkButtonElement(ButtonElement):
     def __init__(
         self,
         *,
-        text: str,
+        text: Union[str, dict, PlainTextObject],
         url: str,
         action_id: Optional[str] = None,
         style: Optional[str] = None,

--- a/tests/slack_sdk/models/test_elements.py
+++ b/tests/slack_sdk/models/test_elements.py
@@ -26,6 +26,7 @@ from slack_sdk.models.blocks import (
     Option,
     InputInteractiveElement,
     InteractiveElement,
+    PlainTextObject,
 )
 from . import STRING_3001_CHARS, STRING_301_CHARS
 
@@ -203,6 +204,23 @@ class LinkButtonElementTests(unittest.TestCase):
             {
                 "text": {"emoji": True, "text": "button text", "type": "plain_text"},
                 "url": "http://google.com",
+                "type": "button",
+                "action_id": button.action_id,
+            },
+            button.to_dict(),
+        )
+
+    # https://github.com/slackapi/python-slack-sdk/issues/1178
+    def test_text_patterns_issue_1178(self):
+        button = LinkButtonElement(
+            action_id="test",
+            text=PlainTextObject(text="button text"),
+            url="http://slack.com",
+        )
+        self.assertDictEqual(
+            {
+                "text": {"text": "button text", "type": "plain_text"},
+                "url": "http://slack.com",
                 "type": "button",
                 "action_id": button.action_id,
             },

